web/utils/permission.ts

```diff
@@ -0,0 +1,18 @@
+import type { DatasetPermission } from '@/models/datasets'
+
+type DatasetConfig = {
+  createdBy: string
+  partialMemberList: string[]
+  permission: DatasetPermission
+}
+
+export const hasEditPermissionForDataset = (userId: string, datasetConfig: DatasetConfig) => {
+  const { createdBy, partialMemberList, permission } = datasetConfig
+  if (permission === 'only_me')
```

Comment:
Since we are defining a new file here, I figure it's better to move those string variables to enums instead.